File: Backend/src/main/java/com/example/backend/VaccineCard.java
package com.example.backend;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class VaccineCard {

    //Properties
    private boolean fullyVaccined;
    private String secondVaccineDate;

    public VaccineCard(){

    }

    public VaccineCard(boolean fullyVaccined, String secondVaccineDate){
        // TO DO: Change mechanism to get vaccine card information from e-devlet
        this.fullyVaccined = fullyVaccined;
        this.secondVaccineDate = secondVaccineDate;
    }
}

File: Frontend/src/classes/VaccineCard.ts
export default class VaccineCard {

    //Properties
    private _fullyVaccined: boolean = false;
    private _secondVaccineDate: string = ""; 

    public constructor (fullyVaccined: boolean , secondVaccineDate: string ){
        // TO DO: Change mechanism to get vaccine card information from e-devlet
        this._fullyVaccined = fullyVaccined;
        this._secondVaccineDate = secondVaccineDate;
    }

    public set fullyVaccined(fullyVaccined: boolean)
    {
        this._fullyVaccined = fullyVaccined;
    }
    public set secondVaccineDate(secondVaccineDate: string)
    {
        this._secondVaccineDate = secondVaccineDate;
    }

    public get fullyVaccined()
    {
        return this._fullyVaccined;
    }
    public get secondVaccineDate()
    {
        return this._secondVaccineDate;
    }
}